### ISSUE TITLE
Code Review: Accessible Content

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/john-travoltage#94]
From https://github.com/phetsims/john-travoltage/issues/89, accessible content in this sim is nearly feature complete.  Before merging with master it would be good to do a code review of new accessibility code.

Since this is a review of just the accessible content, not everything from the 'Code Review Checklist' applies, or needs to be verified at this time.  Some items to the checklist should probably be added for new accessible content!  Here is a modified code review list for this review.  Maybe something like this could eventually go in phet-info, or new items could be added to the already existing list.

PhET code-review checklist
=============

**Build and Run Checks**

- [x] Does the sim pass lint?
- [x] Does the sim build without errors?
- [x] Does the sim start up? (requirejs and built versions)
- [x] Does the sim experience any assertion failures? (run with query parameter 'ea')
- [x] Does the sim pass a scenery fuzzer test? (run with query parameter 'fuzzMouse')

**Strings**
- [x] Are there any unused strings? (Strings that appear in the JSON file but are unused in the simulation should be removed).
- [x] Are there any strings that are not being internationalized? (run with query parameter 'stringTest=x', you should see nothing but 'x' strings, *and hear nothing box 'x' when run with a screen reader*)
- [x] Does the sim layout gracefully handle internationalized strings that are exceptionally long? (run with query parameter 'stringTest=long')
- [x] Does the sim stay on the sim page (doesn't redirect to an external page) when running with the query parameter 'stringTest=xss'. Only test on one desktop platform.
- [x] Make sure the string keys are all perfect, they are difficult to change after publication.

**Coding conventions**

- [x] Does the code style meet PhET's code-style guidelines, as described in the [“Coding Style Guidelines” section of "PhET Development Overview"](https://docs.google.com/document/d/1odXkliRagq0zuf1_NdOtQ2BrkC9hwlISnPi2y-dKdrk/edit#heading=h.1oxr3ptyo50w)?
- [x] Is the code formatted according to PhET conventions? See [phet-idea-code-style.xml](https://github.com/phetsims/joist/blob/master/util/phet-idea-codestyle.xml) for IntelliJ IDEA code style.

**Documentation**

- [x] Are documentation conventions followed, as described in the [“Coding Style Guidelines” section of the PhET Development Overview](https://docs.google.com/document/d/1odXkliRagq0zuf1_NdOtQ2BrkC9hwlISnPi2y-dKdrk/edit#heading=h.1oxr3ptyo50w)?
- [x] Are sim-specific query parameters (if any) identified and documented in one .js file? The .js file should be named <REPO>QueryParameters, for example ArithmeticQueryParameters for the aritmetic repository.

**Common Errors**

- [x] Is `Math.round` used where `dot.Util.roundSymmetric` should be used?  Math.round does not treat positive and negative numbers symmetrically, see https://github.com/phetsims/dot/issues/35#issuecomment-113587879
- [x] Is `toFixed` used where `dot.Util.toFixed` or `dot.Util.toFixedNumber` should be used? JavaScript's `toFixed` is notoriously buggy, behavior differs depending on browser, because the spec doesn't specify whether to round or floor.
- [x] User interface components with `enabled:false` should also have `pickable:false`
- [x] Are random numbers using DOT/Random or seedable sources?

**Organization, Readability, Maintainability**

- [x] Are appropriate design patterns used?
- [x] Is inheritance used where appropriate? Does the type hierarchy make sense?
- [x] Is there any unnecessary coupling? (e.g., by passing large objects to constructors, or exposing unnecessary properties/functions)
- [x] Is there too much unnecessary decoupling? (e.g. by passing all of the properties of an object independently instead of passing the object itself)?
- [x] Are the source files reasonable in size? Scrutinize large files with too many responsibilities - can responsibilities be broken into smaller delegates?
- [x] Are any significant chunks of code duplicated? This will be checked manually as well as with https://github.com/danielstjules/jsinspect
- [x] Is there anything that should be generalized and migrated to common code?
- [x] Are there any TODO or FIXME comments in the code?  They should be addressed or promoted to GitHub issues.

**Memory Leaks**

- [x] Does a heap comparison using Chrome Developer Tools indicate a memory leak? (Describing this process is beyond the scope of this document.)

**Comment by jessegreenberg** [comment]
> Does a heap comparison using Chrome Developer Tools indicate a memory leak? (Describing this process is beyond the scope of this document.)

 Not related to accessibiliyt, but see #97.

**Comment by jessegreenberg** [comment]
> Does the sim pass lint?

Sim will pass lint after completion of #100.

**Comment by jessegreenberg** [comment]
> Does the sim experience any assertion failures? (run with query parameter 'ea')

**Comment by jessegreenberg** [comment]
> Math.round used where dot.Util.roundSymmetric should be used?

Will be fixed by #105.

**Comment by jessegreenberg** [comment]
Code review complete, it looks great and is very clean @jobara, nice work!  If you have any questions about issues created from the review please let me know.

**Comment by jobara** [comment]
@jessegreenberg Thanks for the code review. I've made changes to address the following issues:

- https://github.com/phetsims/john-travoltage/issues/95
- https://github.com/phetsims/john-travoltage/issues/96
- https://github.com/phetsims/john-travoltage/issues/100
- https://github.com/phetsims/john-travoltage/issues/101
- https://github.com/phetsims/john-travoltage/issues/102
- https://github.com/phetsims/john-travoltage/issues/103
- https://github.com/phetsims/john-travoltage/issues/104
- https://github.com/phetsims/john-travoltage/issues/105

I also made a change based on your suggestion from https://github.com/phetsims/john-travoltage/issues/99

**Comment by jessegreenberg** [comment]
Thanks @jobara, everything looks great.  Review complete.
